server/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsDataTO.java

```diff
@@ -0,0 +1,60 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.cloudstack.diagnostics;
+
+import com.cloud.agent.api.to.DataObjectType;
+import com.cloud.agent.api.to.DataStoreTO;
+import com.cloud.agent.api.to.DataTO;
+import com.cloud.hypervisor.Hypervisor;
+
+public class DiagnosticsDataTO implements DataTO {
+    private DataStoreTO dataStoreTO;
+    private Hypervisor.HypervisorType hypervisorType;
+    private String path;
+    private long id;
+
+    public DiagnosticsDataTO(Hypervisor.HypervisorType hypervisorType, DataStoreTO dataStoreTO) {
+        this.hypervisorType = hypervisorType;
+        this.dataStoreTO = dataStoreTO;
+    }
+
+    @Override
+    public DataObjectType getObjectType() {
+        return DataObjectType.TAR;
+    }
+
+    @Override
+    public DataStoreTO getDataStore() {
+        return dataStoreTO;
+    }
+
+    @Override
+    public Hypervisor.HypervisorType getHypervisorType() {
+        return hypervisorType;
+    }
+
+    @Override
+    public String getPath() {
+        return null;
```

Comment:
Same here, you can return path which is already defined as parameter

Reply:
Changed to return parameter